sqlmesh/integrations/github/cicd/command.py

```diff
@@ -166,41 +170,55 @@ def _run_all(
         else:
             raise CICDBotError(f"Unsupported command: {command}")
     controller.update_pr_environment_check(status=GithubCheckStatus.QUEUED)
-    controller.update_prod_environment_check(status=GithubCheckStatus.QUEUED)
+    controller.update_prod_plan_preview_check(status=GithubCheckStatus.QUEUED)
     controller.update_test_check(status=GithubCheckStatus.QUEUED)
+    if is_auto_deploying_prod:
+        controller.update_prod_environment_check(status=GithubCheckStatus.QUEUED)
     tests_passed = _run_tests(controller)
-    if not has_required_approval and controller.do_required_approval_check:
-        controller.update_required_approval_check(status=GithubCheckStatus.QUEUED)
-        has_required_approval = _check_required_approvers(controller)
+    if controller.do_required_approval_check:
+        if has_required_approval:
+            controller.update_required_approval_check(
+                status=GithubCheckStatus.COMPLETED, conclusion=GithubCheckConclusion.SKIPPED
+            )
+        else:
+            controller.update_required_approval_check(status=GithubCheckStatus.QUEUED)
+            has_required_approval = _check_required_approvers(controller)
+    pr_environment_updated = _update_pr_environment(controller)
+    prod_plan_generated = False
+    if pr_environment_updated:
+        prod_plan_generated = _gen_prod_plan(controller)
     else:
-        controller.update_required_approval_check(
+        controller.update_prod_plan_preview_check(
             status=GithubCheckStatus.COMPLETED, conclusion=GithubCheckConclusion.SKIPPED
         )
-    pr_environment_updated = _update_pr_environment(controller)
-    if tests_passed and has_required_approval and pr_environment_updated:
-        _deploy_production(
-            controller, merge_method=merge_method, delete_environment_after_deploy=delete
-        )
-    else:
+    if tests_passed and has_required_approval and pr_environment_updated and prod_plan_generated:
+        _deploy_production(controller)
+    elif is_auto_deploying_prod:
+        if not tests_passed:
+            skip_reason = "Skipped Deploying to Production because unit tests failed"
+        elif not has_required_approval:
+            skip_reason = (
+                "Skipped Deploying to Production because a required approver has not approved"
+            )
+        elif not pr_environment_updated:
+            skip_reason = (
+                "Skipped Deploying to Production because the PR environment was not updated"
+            )
+        elif not prod_plan_generated:
+            skip_reason = (
+                "Skipped Deploying to Production because the production plan could not be generated"
+            )
+        else:
+            skip_reason = "Skipped Deploying to Production for an unknown reason"
```

Comment:
Is it even possible to reach this branch with the current implementation?

Reply:
No there is not a known way to reach this in current implementation.